test/spec/server/request/req.params.ts

```diff
@@ -24,5 +26,48 @@ export default function() {
       expect(params).to.have.property('user')
         .and.that.equals('dolphin');
     });
+
+    it('should decode URI component', async () => {
+      const path = uniquePath();
+      const trackName = 'なんでもないや';
+
+      const { params } = await requestToPromise({
+        route: `${path}/:track`,
+        requestURL: `${path}/${encodeURIComponent(trackName)}`,
+      });
+
+      expect(params.track).to.equal(trackName);
+    });
+
+    it('should throw an error to console', async function () {
+      if (/Edge|Trident/.test(navigator.userAgent)) {
+        // bad URI can't be sent in IE
+        return this.skip();
+      }
+
+      const { router } = createServer();
+      const path = uniquePath();
+      const badURI = '%E0%A4%A';
+
+      let error: any;
+
+      const nativeLog = console.error.bind(console);
+
+      console.error = (...msg) => {
+        error = msg;
+      };
+
+      router.get(`${path}/:id`, (res, req) => {
+        req.send(res.params.id);
+
+        console.error = nativeLog;
+      });
+
+      const { body } = await sendRequest(`${path}/${badURI}`);
+
+      console.log(error);
+      expect(error).not.to.be.undefined;
```

Comment:
To be more specific? `expect(error).to.equal('decode param:  xxx failed)`. Or at least `expect(error).to.include('decode param:  xxx failed)`.